### ISSUE TITLE
Language codenames should be standard and homogeneous

### ISSUE THREAD

**Issue by evilaliv3** [EvanHahn/HumanizeDuration.js#49]
Language codenames should be standard and homogeneous; currently all are in ISO_639-1 except for 'nob' used for Norwegian that is ISO_639-2

My suggestion is that you use [https://en.wikipedia.org/wiki/List_of_ISO_639-1_codes](ISO_639-1)

Using an homogenous naming would make more easy to integrate with other libraries and portals (e.g. [https://www.transifex.com/](Transifex))

In addition for compatibility with most common tools and libraries language codes that include the dialect code _ should be preferred to -: e.g. zh-CN should be written zh_CN (this apply in compatibility with Wordpress, Transifex, libc libraries, etc)

**Comment by EvanHahn** [comment]
I don't know enough about this to make a decision.

A problem I encounter is Chinese. I don't really understand the difference between `zh_CN` and `zh_TW`, but there _is_ a difference (look at the code).

- [ISO 639-1 codes](https://en.wikipedia.org/wiki/List_of_ISO_639-1_codes) only have _one_ code for Chinese. That means we have to use an underscore (`zh_CN`).
- [ISO 639-2 codes](https://en.wikipedia.org/wiki/List_of_ISO_639-2_codes) distinguish between Chinese and the Chinese _family_. I don't know enough about Chinese to know if those conveniently map to the two definitions we have today. If we chose this, we'd also have to choose between B and T.

Pretty sure we cannot use [ISO 639-3 codes](https://en.wikipedia.org/wiki/ISO_639_macrolanguage) because they don't have things like English or Spanish.

I want to get this solved but I don't know enough about this stuff.

**Comment by evilaliv3** [comment]
the difference among zh_CN and zh_TW is the dialect. the first is what is called basic chinese (the more common understandable quite from all chineses) while the latter is the one talked in taiwan.
so when you have languages with multiple dialects it's better to expose the language with language_dialect to clarify what is the one used.
here you can find a more long list: https://github.com/globaleaks/GlobaLeaks/tree/master/client/app/l10n
were for example you can find two dialects for Hungarian, and two for serbian. 
the case of the serbian is really peculiar for you:
sr_RS
sr_RS@latin

this two means two serbian, both for the main generic region but the first is in cyrillic characters, while the latter in latin character.

**Comment by evilaliv3** [comment]
the naming conventions explained before are the one used by transifex.
the same are used by wordpress here: https://packages.debian.org/jessie/all/wordpress-l10n/filelist

**Comment by EvanHahn** [comment]
So would the API look something like this?

```js
humanizeDuration(12345, {
  language: "zh",
  locale: "cn"
})
```

And is there a standard for these pairs? I couldn't find `zh_CN` or `zh_TW` in any official ISO standard.

**Comment by evilaliv3** [comment]
i think we should simple keep a language name "zh_CN" and "zh_TW" and not splitting into language and locale exactly as you have doe so far and listing both in the support. will  be the user that will select directly the dialect by chosing among  "zh_CN" or "zh_TW"

this apply not only to chinese but to every language. in the future for example it will be possible that for portugal (pt) for each now we have simply a generic "pt" we may ave pt_PT (the portugal talked in Portugal) and pt_BR (the portugal talked in Brazil) that are really different (quite two different languages). do you get the point?

**Comment by EvanHahn** [comment]
Closed in 3.0.0.

**Comment by evilaliv3** [comment]
well done man!
